pkg/crinit/aggregated/aggregated_test.go

```diff
@@ -0,0 +1,152 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aggregated
+
+import (
+	"bytes"
+	"testing"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	client "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/cluster-registry/pkg/crinit/common"
+	"k8s.io/cluster-registry/pkg/crinit/options"
+	"k8s.io/cluster-registry/pkg/crinit/util"
+	fakeclientset "k8s.io/kube-aggregator/pkg/client/clientset_generated/clientset/fake"
+)
+
+func TestCreateRBACObjects(t *testing.T) {
+	testCases := []struct {
+		desc        string
+		buffer      *bytes.Buffer
+		opts        *aggregatedClusterRegistryOptions
+		client      client.Interface
+		objExpected bool
+	}{
+		{
+			desc:   "should create RBAC options",
+			buffer: &bytes.Buffer{},
+			opts: &aggregatedClusterRegistryOptions{
+				SubcommandOptions: options.SubcommandOptions{
+					Name: "test1",
+					ClusterRegistryNamespace: "cr-namespace",
+					DryRun: false,
+				},
+			},
+			client:      fake.NewSimpleClientset(),
+			objExpected: true,
+		},
+		{
+			desc:   "dry run should not create RBAC options",
+			buffer: &bytes.Buffer{},
+			opts: &aggregatedClusterRegistryOptions{
+				SubcommandOptions: options.SubcommandOptions{
+					Name: "test2",
+					ClusterRegistryNamespace: "dry-namespace",
+					DryRun: true,
+				},
+			},
+			client:      fake.NewSimpleClientset(),
+			objExpected: false,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
+			rbacObj, err := createRBACObjects(tc.buffer, tc.client, tc.opts)
+			if err != nil {
+				t.Error("Failed to create an RBAC Object")
+			}
+			chkrbac, _ := tc.client.CoreV1().ServiceAccounts(tc.opts.ClusterRegistryNamespace).Get(rbacObj.Name, metav1.GetOptions{})
```

Comment:
I think this test will need to be extended in a later PR to validate the contents of the returned object, and the other objects being created as well: this method creates several resources, and we want to make sure that the created objects match our expectations as to what they will contain.